api/extra.go

```diff
@@ -49,14 +55,102 @@ func ResolveDidToHandle(ctx context.Context, xrpcc *xrpc.Client, res did.Resolve
 		return "", "", fmt.Errorf("our XRPC client is authed for a different pds (%s != %s)", svc.ServiceEndpoint, xrpcc.Host)
 	}
 
-	verdid, err := comatproto.IdentityResolveHandle(ctx, xrpcc, handle)
+	verdid, err := hr.ResolveHandleToDid(ctx, handle)
 	if err != nil {
 		return "", "", err
 	}
 
-	if verdid.Did != udid {
+	if verdid != udid {
 		return "", "", fmt.Errorf("pds server reported different did for claimed handle")
 	}
 
 	return handle, svc.ServiceEndpoint, nil
 }
+
+type HandleResolver interface {
```

Comment:
feel like this should go somewhere other than `api/extra.go`... maybe in util, or a new top-level package? but not urgent, can live here for now.

Reply:
yeah, it might deserve its own package once we feel it out a bit more. For now though all users were previously using the api package for the old behavior, so this is 'drop in' in terms of deps